### PR TITLE
The Great Tweaks Update of July 2020

### PR DESCRIPTION
**Pull request by hobinjk**
Not sure whether it's better to do this or have separate options for new asks/new posts/new messages/new activity but I do know that this way is easier to implement.

### CODE REVIEW

File: Extensions/tweaks.js [hunk 1]
```diff
@@ -726,8 +741,20 @@ XKit.extensions.tweaks = new Object({
...
+			XKit.extensions.tweaks.add_css(`${notificationBadgeSel} {
+				display: none !important;
+			}`, 'xkit_tweaks_hide_notification_badges');
+		}
 
+		if (XKit.extensions.tweaks.preferences.hide_post_highlight.value) {
+			const ltoSel = XKit.css_map.keyToCss('listTimelineObjectInner');
+			XKit.extensions.tweaks.add_css(`${ltoSel} {
+				box-shadow: none !important;
+			}`, 'xkit_tweaks_hide_notification_badges');
```
Comment (nightpool) [inline Extensions/tweaks.js:755]: should be `xkit_tweaks_hide_post_highlights`

File: Extensions/tweaks.js [hunk 2]
```diff
@@ -296,7 +296,19 @@ XKit.extensions.tweaks = new Object({
 			default: false,
 			value: false,
 			desktop_only: true
-		}
+		},
+		"hide_notification_badges": {
```
Comment (nightpool) [inline Extensions/tweaks.js:300]: no quotes?

**Comment by hobinjk** [comment]
@nightpool Updated, thanks! I think an icon would be useful but adding it would take away from fixing things (although it probably wouldn't take much to be honest)

**Comment by invalidCards** [comment]
Things I still want to add to this:
- Drop-down box to select notification badge style. Choices are default (no CSS applied), hidden (currently PR'd CSS applied), white, and Old Red (red background with white text)
- A tweak to hide *just* the activity notification badge, as a lot of people have requested that recently

**Comment by hobinjk** [comment]
👍 go ahead :)
https://www.youtube.com/watch?v=YZvbCgnCJoA

File: Extensions/tweaks.js [hunk 3]
```diff
@@ -726,8 +767,42 @@ XKit.extensions.tweaks = new Object({
...
+		if (XKit.extensions.tweaks.preferences.hide_post_highlight.value) {
+			const ltoSel = XKit.css_map.keyToCss('listTimelineObjectInner');
+			XKit.extensions.tweaks.add_css(`${ltoSel} {
+				box-shadow: none !important;
+			}`, 'xkit_tweaks_hide_post_highlight');
+		}
+		
+		if (XKit.extensions.tweaks.preferences.grayscale_new_post_button.value) {
+			let postIconButtonSel = XKit.css_map.keyToCss('postIconButton');
+			XKit.extensions.tweaks.add_css(`${postIconButtonSel} span { filter: grayscale(100%); }`, 'xkit_tweaks_grayscale_new_post_button');
```
[the remark this answers is not in the document]
Comment (nightpool) [reply Extensions/tweaks.js:802]: descendantSelector won't work here b/c they're not both keys but this should work:

```js
  const postIconButtonSel = XKit.css_map.keyToClasses('postIconButton').map(class => `${class} span`).join(',');
```

**Pull request merged**